python/ray/tests/test_signal_handler.py

```diff
@@ -0,0 +1,204 @@
+import asyncio
+import os
+import signal
+import sys
+import tempfile
+import textwrap
+import time
+
+import pytest
+
+import ray
+from ray._common.test_utils import wait_for_condition
+from ray._private.test_utils import run_string_as_driver_nonblocking
+
+
+@ray.remote
+class SimpleActor:
+    def pid(self) -> int:
+        return os.getpid()
+
+    def ping(self) -> str:
+        return "ok"
+
+
+@ray.remote
+class AsyncioActor:
+    async def pid(self) -> int:
+        return os.getpid()
+
+    async def ping(self) -> str:
+        return "ok"
+
+    async def run(self):
+        while True:
+            await asyncio.sleep(0.1)
+
+
+def _expect_actor_dies(actor: "ray.actor.ActorHandle", timeout_s: float = 30) -> None:
+    """Wait for an actor to die and raise if it doesn't die within timeout.
+
+    Args:
+        actor: Ray actor handle to monitor.
+        timeout_s: Maximum time to wait for actor death in seconds.
+
+    Raises:
+        AssertionError: If actor doesn't die within timeout_s.
+    """
+    start = time.monotonic()
+    while time.monotonic() - start < timeout_s:
+        try:
+            ray.get(actor.ping.remote(), timeout=1)
+        except ray.exceptions.GetTimeoutError:
+            continue
+        except ray.exceptions.RayActorError:
+            return
+        except Exception:
+            return
+    raise AssertionError("Actor did not die within timeout")
+
+
+def test_asyncio_actor_force_exit_is_immediate(ray_start):
+    """Calling force_exit_worker() inside an asyncio actor should exit immediately."""
```

Comment:
this is testing against internal implementation details, not user-facing APIs/behaviors